### PR TITLE
Update popover modal type

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
There's a bit of history to this PR. Basically, I'm:

1) Extracting some changes to simplify [this PR](https://github.com/Expensify/Expensify.cash/pull/1469/files), and
2) Taking a conclusion from [this slack thread](https://expensify.slack.com/archives/C011W8BJ9L6/p1613725676116500), namely "we have a separate, established rule for popovers. If the screen is small (mobile), the popover is bottom-docked. If the screen is wide, launch popover near where the click/press event happened", and put it into code.

Ultimately this is a prep step for the ReportActionContextMenu. 

### Fixed Issues (parital)
https://github.com/Expensify/Expensify/issues/147472

### Tests
For now, we're just going to perform a regression test of the existing popover modal.

1. Run the app.
2. Click on the global create floating action button
3. Ensure that the _correct type_ of popover modal appears.

On wide screens, it should look like this (image taken from production):

On narrow screens, it should look like this (image taken from production):

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots

|Platform|Wide|Narrow|
|---|---|---|
|Web|image|image|
|mWeb|n/a|image|
|Desktop|image|image|
|iOS|n/a|image|
|Android|n/a|image|

### CODE REVIEW

**Review by yuwenmemon — COMMENTED** [review]
Changes seem fine to me. Curious to see what @marcaaron says.

Have we considered creating a `PopoverModal` component to wrap `BaseModal`? Then you could do both the adjustment of styling based on screen size and handling of anchor position in that file instead of mixing it in with the more generalized Modal code, right?

File: src/styles/getModalStyles.js [hunk 1]
```diff
@@ -18,7 +18,11 @@ export default (type, windowDimensions) => {
     let shouldAddBottomSafeAreaPadding = false;
     let shouldAddTopSafeAreaPadding = false;
 
-    switch (type) {
+    const platformAdjustedType = windowDimensions.isSmallScreenWidth && type === CONST.MODAL.MODAL_TYPE.POPOVER
```
Comment (marcaaron) [inline src/styles/getModalStyles.js:21]: Not really "platform", more like `screenWidthAdjustedType`

**Review by marcaaron — CHANGES_REQUESTED** [review]
Looks good just have one small comment

**Comment by marcaaron** [comment]
>Have we considered creating a PopoverModal component to wrap BaseModal? Then you could do both the adjustment of styling based on screen size and handling of anchor position in that file instead of mixing it in with the more generalized Modal code, right?

I'd also be open to seeing that. It makes sense to me and we could just call it `Popover`. Thinking about it some more it does seem strange that if you ask to use one modal type it gets switched to another modal type.

**Comment by roryabraham** [comment]
Okay, I'll create a Popover component.

**Comment by roryabraham** [comment]
Actually, thinking about this some more, I'm not sure I see a clean way to do that.

> Then you could do ... the handling of anchor position in that file.

The problem I see with this is that the modal styles are all generated internally in the `BaseModal` component, so creating a wrapper we have no good way to edit those styles without creating props to override certain internal styles. My first implementation of a related PR took that approach, and I think we all agreed it was too "anything goes" and a divergence from our clearly-defined modal system.

> Thinking about it some more it does seem strange that if you ask to use one modal type it gets switched to another modal type

Maybe so, but that's exactly what _should_ happen to establish this premise:

> we have a separate, established rule for popovers. If the screen is small (mobile), the popover is bottom-docked. If the screen is wide, launch popover near where the click/press event happened

Right?

---

Also, FWIW, I plan on creating a wrapper component called `PopoverWithMeasuredPosition`, which would anchor the modal position onto a wrapped element internally.

File: src/components/Modal/ModalPropTypes.js [hunk 2]
```diff
@@ -25,7 +25,23 @@ const modalPropTypes = {
...
+
     ...windowDimensionsPropTypes,
 };
 
-export default modalPropTypes;
+const defaultProps = {
+    onSubmit: null,
+    type: '',
+    onModalHide: () => {},
+    popoverAnchorPosition: null,
```
[the remark this answers is not in the document]
Comment (roryabraham) [reply src/components/Modal/ModalPropTypes.js]: Updated to an empty object

**Comment by marcaaron** [comment]
>Maybe so, but that's exactly what should happen to establish this premise: we have a separate, established rule for popovers. If the screen is small (mobile), the popover is bottom-docked. If the screen is wide, launch popover near where the click/press event happened Right?

I could go either way on this. But it might help to think of `Modal` as one construct able to take "type" instructions and produce several different designs. It has a "type" and if you give it that "type" that's what you'd expect to get back. If only one of those types needs to "change if we have a different window width" but the rest of them don't - that could possibly be a sign that we should encapsulating the exceptional behavior into another construct like `Popover` or maybe more accurately `ResponsivePopover` or something.

Anyways, I don't think you need to get too loose with it. It would basically just be

```jsx
const Popover = props => (
    <Modal 
        type={props.smallWidth ? type1 : type2}
        popoverAnchorDimensions={props.anchorDimensions}
    />
);
```

**Comment by roryabraham** [comment]
Okay, updated to create the `Popover` component.

**Review by marcaaron — APPROVED** [review]
Thanks for the changes. Looks good and tests well.

**Review by yuwenmemon — APPROVED** [review]

**Pull request merged**